cmake/modules/AddSwift.cmake

```diff
@@ -1779,14 +1862,17 @@ function(_add_swift_executable_single name)
         "-Xlinker" "@executable_path/../lib/swift/${SWIFT_SDK_${SWIFTEXE_SINGLE_SDK}_LIB_SUBDIR}")
   endif()
 
-  if(SWIFT_ENABLE_GOLD_LINKER AND
-     "${SWIFT_SDK_${SWIFTEXE_SINGLE_SDK}_OBJECT_FORMAT}" STREQUAL "ELF")
-    list(APPEND link_flags "-fuse-ld=gold")
-  endif()
-  if(SWIFT_ENABLE_LLD_LINKER OR
-     ("${SWIFTLIB_SINGLE_SDK}" STREQUAL "WINDOWS" AND
-      NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "WINDOWS"))
-    list(APPEND link_flags "-fuse-ld=lld")
+  # MSVC doesn't recognise -fuse-ld
+  if (NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    if(SWIFT_ENABLE_GOLD_LINKER AND
+       "${SWIFT_SDK_${SWIFTEXE_SINGLE_SDK}_OBJECT_FORMAT}" STREQUAL "ELF")
+      list(APPEND link_flags "-fuse-ld=gold")
+    endif()
+    if(SWIFT_ENABLE_LLD_LINKER OR
+       ("${SWIFTLIB_SINGLE_SDK}" STREQUAL "WINDOWS" AND
+        NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "WINDOWS"))
+      list(APPEND link_flags "-fuse-ld=lld")
+    endif()
```

Comment:
See the comment 2 above:
How would MSVC understand -fuse-ld=lld even if we're cross-compiling to Linux?

I've changed this code BTW to:

  # MSVC doesn't recognise -fuse-ld.
  if ("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
    if(SWIFT_ENABLE_GOLD_LINKER AND
       "${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_OBJECT_FORMAT}" STREQUAL "ELF")
      list(APPEND link_flags "-fuse-ld=gold")
    endif()
    if(SWIFT_ENABLE_LLD_LINKER OR
       ("${SWIFTLIB_SINGLE_SDK}" STREQUAL "WINDOWS" AND
        NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "WINDOWS"))
      list(APPEND link_flags "-fuse-ld=lld")
    endif()
  endif()
Perhaps this addresses the cross-compilation concerns?